File: Models/Tierlist.cs
namespace ranking_app.Models;

public class TierlistModel
{
    public Guid Id { get; set; }
    public required string Name { get; set; }
    public Guid TemplateId { get; set; }
    public string? UserId { get; set; }
    public required ICollection<RankedElement> RankedElements { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: ClientApp/src/models/Tierlist.ts
import { RankedElement } from './Element';

export interface Tierlist {
    id: string;
    name: string;
    userId?: string;
    templateId: string;
    rankedElements: RankedElement[];
    createdAt: string; // ex: "2023-11-16T20:26:59.654363Z"
}

export enum ETierlistDragItem {
    Element = 'element'
}
